### PR TITLE
MM-11270: Forbid react/de-react in archived channels

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Forbid to react and remove reactions in archived channels.

#### Ticket Link
[MM-11270](https://mattermost.atlassian.net/browse/11270)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-server/blob/master/i18n/en.json)) updates

### CODE REVIEW

File: api4/reaction_test.go [hunk 1]
```diff
@@ -192,13 +192,36 @@ func TestSaveReaction(t *testing.T) {
 		_, resp := Client.SaveReaction(reaction)
 		CheckForbiddenStatus(t, resp)
 
-		if reactions, err := th.App.GetReactionsForPost(postId); err != nil || len(reactions) != 3 {
-			t.Fatal("should have not created a reactions")
+		if reactions, err := th.App.GetReactionsForPost(post.Id); err != nil || len(reactions) != 0 {
```
Comment (jespino) [inline api4/reaction_test.go]: This was a bug in the previous exisiting test, not related to my changes.

File: app/reaction.go [hunk 2]
```diff
@@ -15,43 +15,46 @@ func (a *App) SaveReactionForPost(reaction *model.Reaction) (*model.Reaction, *m
...
 
-	if a.License() != nil && *a.Config().TeamSettings.ExperimentalTownSquareIsReadOnly {
-		var channel *model.Channel
-		if channel, err = a.GetChannel(post.ChannelId); err != nil {
+	channel, err := a.GetChannel(post.ChannelId)
+	if err != nil {
+		return nil, err
+	}
+
+	if channel.DeleteAt > 0 {
```
Comment (jespino) [inline app/reaction.go]: This is the main change (and other exactly equal in other function), the rest of the code is transformed from normal error handling to idiomatic error handling.

File: app/reaction.go [hunk 3]
```diff
@@ -60,21 +63,23 @@ func (a *App) DeleteReactionForPost(reaction *model.Reaction) *model.AppError {
...
 
-	if a.License() != nil && *a.Config().TeamSettings.ExperimentalTownSquareIsReadOnly {
-		var channel *model.Channel
-		if channel, err = a.GetChannel(post.ChannelId); err != nil {
+	channel, err := a.GetChannel(post.ChannelId)
+	if err != nil {
+		return err
+	}
+
+	if channel.DeleteAt > 0 {
```
Comment (jespino) [inline app/reaction.go]: This is the main change (and other exactly equal in other function), the rest of the code is transformed from normal error handling to idiomatic error handling.

File: api4/reaction_test.go [hunk 4]
```diff
@@ -507,10 +530,40 @@ func TestDeleteReaction(t *testing.T) {
...
 		if reactions, err := th.App.GetReactionsForPost(postId); err != nil || len(reactions) != 1 {
-			t.Fatal("should have not deleted a reactions")
+			t.Fatal("should have not deleted a reaction")
 		}
 
 		th.App.RemoveLicense()
 		th.App.UpdateConfig(func(cfg *model.Config) { *cfg.TeamSettings.ExperimentalTownSquareIsReadOnly = false })
 	})
+
+	t.Run("unable-to-deletect-reactions-in-an-archived-channel", func(t *testing.T) {
```
Comment (saturninoabril) [inline api4/reaction_test.go]: Minor typo --> `deletect`

**Pull request merged**